lib/utils/fixEmptyLineBefore.mjs

```diff
@@ -0,0 +1,39 @@
+import { assert } from './validateTypes.mjs';
+
+/** @typedef {import('stylelint').Range} Range */
+/** @typedef {import('postcss').Node} Node */
+/** @typedef {(node: Node, newline: string) => Node} Callback */
+
+/**
+ * fix callback for *-empty-line-before rules
+ * @param {object} o
+ * @param {Node} o.node
+ * @param {string=} o.newline
```

Comment:
[ask] Typo?

```suggestion
 * @param {string} o.newline
```

Reply:
No it's optional.
There are two ways supported in JSDOC.
Since a refactor—that we talked about—is planned which will change `newline` to be never `undefined`, it won't be missed.

Reply:
Then, I prefer `string | undefined` because it's explicit. I believe it's more readable for many people.

Reply:
But it's fine for now since it will be refactored.